rmgpy/rmg/main.py

```diff
@@ -1512,7 +1547,119 @@ def initializeLog(verbose, log_file_name):
     logger.addHandler(fh)
 
 ################################################################################
+class RMG_Memory:
+    """
+    class for remembering RMG simulations
+    and determining what simulation to run next
+    """
+    def __init__(self,reactionSystem):
+        self.Ranges = dict()
+        if hasattr(reactionSystem,'Trange') and type(reactionSystem.Trange) == list:
+            Trange = reactionSystem.Trange
+            self.Ranges['T'] = [T.value_si for T in Trange]
+        if hasattr(reactionSystem,'Prange') and type(reactionSystem.Prange) == list:
+            Prange = reactionSystem.Prange
+            self.Ranges['P'] = [np.log(P.value_si) for P in Prange]
+        if hasattr(reactionSystem,'initialMoleFractions'):
+            for key,value in reactionSystem.initialMoleFractions.iteritems():
+                assert key != 'T' and key != 'P', 'naming a species T or P is forbidden'
+                if type(value) == list:
+                    self.Ranges[key] = value
+        if hasattr(reactionSystem,'initialConcentrations'):
+            for key,value in reactionSystem.initialConcentrations.iteritems():
+                assert key != 'T' and key != 'P', 'naming a species T or P is forbidden'
+                if type(value) == list:
+                    self.Ranges[key] = value.value_si
+                    
+        for term in reactionSystem.termination:
+            if isinstance(term, TerminationTime):
+                self.tmax = term.time.value_si
+        
+        self.normalize = hasattr(reactionSystem,'initialMoleFractions')
+        self.conditionList = []
+        self.scaledConditionList = []
+        self.ts = []
+        self.convs = []
+        self.Ns = []
+        
+    def addtConvN(self,t,conv,N):
```

Comment:
Please change new function names to lowercase

Reply:
done